### PR TITLE
Refactor base controller,	add internet-radio.com

### PR DESCRIPTION
**Pull request by garmoshka-mo**

### CODE REVIEW

**Comment by berrberr** [comment]
@garmoshka-mo Thanks for this PR. I like most of the changes. I need to do a more thorough review since there are some pretty big changes here :)

File: code/js/modules/BaseController.js [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
       //** Song Info **//
-      song: (options.song || null),
-      artist: (options.artist || null)
-    };
+      'song', 'artist'
+    ];
+    selectorsKeys.forEach((key) => this.selectors[key] = null);
+
+    // Assign passed options to this and this.selectors
+    Object.keys(options).forEach((key) => {
```
Comment (egeste) [inline code/js/modules/BaseController.js]: This project already has `lodash` as a dependency. es6 is cool and all, but this function could more or less be 2 lines of lodash operations - and inherently more reliable as there's no custom code to test

File: code/js/controllers/InternetRadioController.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  var BaseController = require("BaseController");
+
+  new BaseController({
+    siteName: "Internet Radio",
+
+    play() { this.launchPlayback(0); },
+    playNext() { this.launchPlayback(1); },
+    playPrev() { this.launchPlayback(-1); },
+
+    pause() {
```
Comment (egeste) [inline code/js/controllers/InternetRadioController.js]: I'm missing something. Why override the functions when you can use the standard selector bindings?

File: code/js/modules/BaseController.js [hunk 3]
```diff
@@ -91,12 +114,20 @@
...
+        this.pause();
+      else
+        this.play();
+
+    } else if(this.selectors.play !== null && this.selectors.pause !== null) {
+
+      if(this.isPlaying())
         this.click({action: "playPause", selectorButton: this.selectors.pause, selectorFrame: this.selectors.iframe});
-      } else {
+      else
```
Comment (egeste) [inline code/js/modules/BaseController.js]: Mixed syntax. Every other conditional in this repo is
```javascript
if () { ... } else { ... }
```

Code quality requires consistency

File: code/js/modules/Sitelist.js [hunk 4]
```diff
@@ -374,7 +375,9 @@
    * @return {Boolean} true if url matches a music site
    */
   Sitelist.prototype.checkMusicSite = function(url) {
-    var sitesRegex = _.map(this.sites, function(site) { return site.urlRegex; });
+    var sitesRegex = _.map(this.sites, function(site) {
+      return site.urlRegex; // separate line for debug ability
```
Comment (egeste) [inline code/js/modules/Sitelist.js]: +1, but remove the comment

File: code/js/modules/utils.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+(function() {
```
Comment (egeste) [inline code/js/modules/utils.js]: Again, lodash

File: karma.conf.js [hunk 6]
```diff
@@ -36,7 +36,7 @@ module.exports = function(config) {
 
     browserify: {
       debug: true,
-      transform: []
+      transform: [ 'babelify' ]
```
Comment (egeste) [inline karma.conf.js]: Adding an entire language dependency to someone else's repo?

**Comment by egeste** [comment]
Overall, good concepts here. I think the addition of `code/js/controllers/InternetRadioController.js` should happen in a separate commit, though.

**Comment by berrberr** [comment]
Sorry, branch too stale now

**Pull request closed without merging**